### ISSUE TITLE
File not found error with Hotpep

### ISSUE THREAD

**Issue by jaube** [linnabrown/run_dbcan#38]
Hello,

I would like to use dbCAN2 to annotate proteins sequences. I have an error about Hotpep in my log file:
> Traceback (most recent call last):
>   File "/home1/datahome/jaube/.local/bin/parallel_group_many_proteins_many_patterns_noDNA.py", line 92, in <module>
>     p.ec = open("%s/%s/%s_group_ec.txt"%(peptide_dir_name, fam, fam)) \
> FileNotFoundError: [Errno 2] No such file or directory: '/home1/datahome/jaube/.local/lib/python3.8/site-packages/Hotpep/CAZY_PPR_patterns/GH/GH149/GH149_group_ec.txt'
> Assigning proteins to groups
> Collecting Results
> Assigning proteins to groups
> Collecting Results
> Traceback (most recent call last):
>   File "/home1/datahome/jaube/.local/bin/parallel_group_many_proteins_many_patterns_noDNA.py", line 92, in <module>
>     p.ec = open("%s/%s/%s_group_ec.txt"%(peptide_dir_name, fam, fam)) \
> FileNotFoundError: [Errno 2] No such file or directory: '/home1/datahome/jaube/.local/lib/python3.8/site-packages/Hotpep/CAZY_PPR_patterns/PL/PL35/PL35_group_ec.txt'
> Assigning proteins to groups
> Collecting Results
> Traceback (most recent call last):
>   File "/home1/datahome/jaube/.local/bin/parallel_group_many_proteins_many_patterns_noDNA.py", line 92, in <module>
>     p.ec = open("%s/%s/%s_group_ec.txt"%(peptide_dir_name, fam, fam)) \
> FileNotFoundError: [Errno 2] No such file or directory: '/home1/datahome/jaube/.local/lib/python3.8/site-packages/Hotpep/CAZY_PPR_patterns/GT/GT106/GT106_group_ec.txt'
> Assigning proteins to groups
> Collecting Results
> Traceback (most recent call last):
>   File "/home1/datahome/jaube/.local/bin/parallel_group_many_proteins_many_patterns_noDNA.py", line 92, in <module>
>     p.ec = open("%s/%s/%s_group_ec.txt"%(peptide_dir_name, fam, fam)) \
> FileNotFoundError: [Errno 2] No such file or directory: '/home1/datahome/jaube/.local/lib/python3.8/site-packages/Hotpep/CAZY_PPR_patterns/CBM/CBM85/CBM85_group_ec.txt'

Also in the output from the Hotpep run I didn't find any protein from the GH, PL, GT or CBM groups while the annotation with hmmer and diamond found many.

**Comment by linnabrown** [comment]
It seems you are missing /home1/datahome/jaube/.local/lib/python3.8/site-packages/Hotpep/CAZY_PPR_patterns/*

Could you `cd /home1/datahome/jaube/.local/lib/python3.8/site-packages/Hotpep/CAZY_PPR_patterns/`(which is the ppr for hotpep) and show me the files under this directory.

**Comment by jaube** [comment]
The attached file contains the output of the following command:
`ls -R /home1/datahome/jaube/.local/lib/python3.8/site-packages/Hotpep/CAZY_PPR_patterns > CAZY_PPR_patterns_files.txt`
The files not found that are named in the error message are not present in the CAZY_PPR_patterns folder of your github.
[CAZY_PPR_patterns_files.txt](https://github.com/linnabrown/run_dbcan/files/4394562/CAZY_PPR_patterns_files.txt)

**Comment by HaidYi** [comment]
I see, we will investigate this problem and give you the feedback ASAP. Thank you for using our tool.

**Comment by HaidYi** [comment]
Hi Dr. Yin @yinlabniu , these are the files which are missing in PPR. 
```
CAZY_PPR_patterns/GH/GH149/GH149_group_ec.txt
CAZY_PPR_patterns/PL/PL35/PL35_group_ec.txt
CAZY_PPR_patterns/GT/GT106/GT106_group_ec.txt
CAZY_PPR_patterns/CBM/CBM85/CBM85_group_ec.txt
```

**Comment by Neato-Nick** [comment]
hotpep output for me has GH, PL, CE, and AA, so I'm missing GTs and CBMs.

Looking at my ppr directories, I noticed I'm mostly missing _group_ec.txt in those two families with no output. Below are the PPR families that are missing a _group_ec.txt file

* CBM82
* CBM84
* CBM85
* GT105
* GT106
* GT107
* PL31
* PL33
* PL34
* PL35
* PL36
* PL37

Weirdly I'm also missing group_ec files for a lot of GH families, but my output contains plenty of those

* GH146-GH154
* GH156
* GH158
* GH159
* GH160-GH165

**Comment by yinlabniu** [comment]
Another user raised the same issue a few weeks ago and we are working to update the PPR data. PPR does not work on CBMs per their paper, but we should have GTs included. It may be that you input happened to not contain GTs?

Yanbin

________________________________

**Comment by Neato-Nick** [comment]
>  It may be that you input happened to not contain GTs?

Totally possible but unlikely. Out of 2291 rows in my cazy overview output file, 713 had at least one "GT" math in either the HMM or DIAMOND columns, 185 rows found a GT using *both* HMM and DIAMOND so those I would consider pretty high confidence. 

Incidentally this is also the same dataset I mentioned in my GT2 issue 
https://github.com/linnabrown/run_dbcan/issues/39#issue-596708974

**Comment by Neato-Nick** [comment]
Actually, is there a small test dataset bundled with this program? I'd be willing to run some proteins guaranteed to return a mix of GT results and check the output.

**Comment by yinlabniu** [comment]
If there are 185 rows having GTs with hmm+diamond, then it must be hotpep not finding them as GTs due to the problem with PPR patterns. It is very surprising that hotpep didn't find any GTs in your query data. Do you not have any hotpep predicted GTs or just no _group_ec.txt files (this is the problem the other user also had)?

**Comment by Neato-Nick** [comment]
Surprisingly, most of the GTs do have _group_ec.txt files. I just
double-checked, the only families missing in the GT are those three I
listed above. So hotpep just didn't find the GTs, maybe I need to run
hotpep separately, outside of dbcan, to double check this?
- Nick

**Comment by yinlabniu** [comment]
Yes, it is a good idea to run hotpep separately and see if it can find the GTs. Will you also send us your input seq file? We would also run on our end to see what's going on.

Thanks,

Yanbin

________________________________

**Comment by Neato-Nick** [comment]
I used run_dbcan.py with --tools hotpep. It crashed pretty quickly with the error below ($HOME actually did show the abs path to my home dir)

`FileNotFoundError: [Errno 2] No such file or directory: '$HOME/.local/lib/python3.7/site-packages/Hotpep/CAZY_PPR_patterns/GT/GT105/GT105_group_ec.txt'`

I emailed you the proteins I've been working with

Thanks!

**Comment by Neato-Nick** [comment]
For others who find this thread: If hotpep classifies a query protein into a family missing a group_ec.txt file, it will not print any output for that superfamily. In my case, hotpep classified a protein as GT105, the error above shows that this family was missing a group_ec.txt file, and that crashed it for the GT superfamily only.

**Comment by chassenr** [comment]
Hi,
just FYI, I have the same problem working with the latest version of dbcan. When I look at the CAZY_PPR_patterns directory, the *_group_ec.txt files seem to be missing for the following 35 families: AA14,AA15,AA16,CBM82,CBM84,CBM85,GH146,GH147,GH148,GH149,GH150,GH151,GH152,GH153,GH154,GH156,GH158,GH159,GH160,GH161,GH162,GH163,GH164,GH165,GT105,GT106,GT107,PL28,PL29,PL31,PL33,PL34,PL35,PL36,PL37.

Cheers,
Christiane

**Comment by acbellorib** [comment]
Hi there,
just adding some information, I'm using dbCAN2 (version 2.0.6) and, after carrying out a similar verification as of Christiane's, I can confirm that the same 35 families are missing regarding the *_group_ec.txt files. The missing file error (GH153_group_ec.txt, for instance) happens when using the EscheriaColiK12MG1655 data example. Apparently, as far as I've inspected, some of these files are also originally missing in the tar package hotpep-python-08-20-2019.tar.gz. Possibly rolling back to the contents of an older dated tar package would do the trick to solve the issue? Thanks for releasing the tool and for your kind attention!

Cheers,
Antonio

**Comment by linnabrown** [comment]
Hi All,

This problem has been solved by adding more ec files. 
We added one more column called "EC number" in hotpep.out file.
If the subfamily does not found out, it will show "NA", otherwise show "x.x.x.x"

Using this command for updating our run_dbcan package, you don't need to install those dependencies again.

Thanks a lot for your feedback!

```
pip install run-dbcan==2.0.8 --user
```

Best,
Le
